src/data-access/MwApiLangCodeRetriever.ts

```diff
@@ -0,0 +1,21 @@
+import { MwApi } from '@/@types/mediawiki';
+import LangCodeRetriever from './LangCodeRetriever';
+import { processWbGetClaimsResponse, WbGetClaimsResponse } from './apiBaseLangCodeRetriever';
+
+export default class MwApiLangCodeRetriever implements LangCodeRetriever {
+
+	public constructor( private api: MwApi, private languageCodeProperty: string ) {
+	}
+
+	public async getLanguageCodeFromItem( itemId: string ): Promise<string | null | false> {
+		// TODO handle errors
```

Comment:
I feel we can do _something_ here to handle errors. Probably would do a try-catch around the `await this.api.get(...)` and in the catch clause log the error to `console.error()` (maybe with a prefix or something?) and then return false, as to let the App recover from, for example, network problems.

Thoughts?

Reply:
I would implement that as a `.catch()` call instead of `try/catch` syntax, so that we can access all of the values that jQuery promises can reject with (for API errors, the first value is the least useful one). Something like:

```js
.catch( function () {
    console.error( 'API error while getting language code', arguments );
    return false;
} );
```

(Maybe also `console.warn()` instead of `console.error()`, not sure…) But in general that sounds okay to me.

Reply:
sounds good, will do!

Reply:
Ok, I have a version up that I think should work